src/structs/mod.rs

```diff
@@ -1,6 +1,49 @@
 use std::fmt;
 use na::DVector;
 
+
+/// Struct for u8
+pub struct Data {
+    /// classifier for the NN (Result)
+    classifier: u8,
```

Comment:
Just to be clear: Implementing it this way means the Data never really knows what type of classifier it's holding